akka-actor-tests/src/test/scala/akka/util/ConfigValidationSpec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.util
+
+import akka.actor.DefaultSupervisorStrategy
+import akka.testkit.AkkaSpec
+import com.typesafe.config.ConfigFactory
+
+class ConfigValidationSpec extends AkkaSpec {
+  import ConfigFactory.parseString
+  import ConfigValidation._
+
+  "ConfigValidation" must {
+    val empty = ConfigFactory.empty
+    val config = system.settings.config
+    val emptyConfigValue = parseString("""akka.empty = "" """)
+
+    "fail fast in validation if an expected plugin is missing" in {
+      intercept[ConfigValidationError](config.plugin("akka.missing.fqcn"))
+      intercept[ConfigValidationError](config.plugin(null))
+      intercept[ConfigValidationError](config.plugin(""))
+      intercept[ConfigValidationError](empty.plugin("akka.missing.fqcn"))
+    }
+
+    "return valid plugin if config is valid" in {
+      val plugin = config.plugin("akka.actor.guardian-supervisor-strategy")
```

Comment:
This is an example of all the plugin boilerplate in the code.
It could be a one-liner and return the `Class`. I just did `Class.forName` as sample. Originally I had a PluginConfigError - note they all extend `IllegalArgumentException` and yes the few that are caught in one function that are real Config exceptions should be those but that's an impl detail to fix and expose. But I simplified to one.